src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -37,226 +39,353 @@
+
+		case DeleteEventCommand.COMMAND_WORD:
+			return prepareDeleteEvent(arguments);
+
+		case EditEventCommand.COMMAND_WORD:
+			return new EditEventParser().parse(arguments);
+
+		case DeleteDeadlineCommand.COMMAND_WORD:
+			return prepareDeleteDeadline(arguments);
+
+		case EditDeadlineCommand.COMMAND_WORD:
+			return new EditDeadlineParser().parse(arguments);
+
+		case ClearCommand.COMMAND_WORD:
+			return new ClearCommand();
+
+		case FindCommand.COMMAND_WORD:
+			return prepareFind(arguments);
+
+		case ListCommand.COMMAND_WORD:
+			return new ListCommand();
+
+		case ExitCommand.COMMAND_WORD:
+			return new ExitCommand();
+
+		case HelpCommand.COMMAND_WORD:
+			return new HelpCommand();
+
+		default:
+			return new IncorrectCommand(MESSAGE_UNKNOWN_COMMAND);
+		}
+	}
+
+	/**
+	 * Parses arguments in the context of the add Task command.
+	 *
+	 * @param args
+	 *            full command args string
+	 * @return the prepared command
+	 */
+	private Command prepareAdd(String args) {
+		args = args.trim();
+		final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(args.trim());// Todos:
+																				// replace
+																				// with
+																				// TASK_DATA_ARGS_FORMAT
+		// Validate arg string format
+		if (!matcher.matches()) {
+			return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+		}
+		String type = idTaskType(args);
+		switch (type) {
+		case ("float"):
+			return new AddFloatingTaskParser().parse(args);
+
+		case ("deadline"):
+			return new AddDeadlineParser().parse(args);
+
+		case ("event"):
+			return new AddEventParser().parse(args);
+		}
+		return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+	}
+
+	/**
+	 * Identifies the type of task.
+	 * 
+	 * @param args
+	 * @return type of task as String
+	 */
+	private String idTaskType(String args) {
+
+		// float, no date or time
+		// deadline, has date or/and time
+		// event has date and time and to keyword
+		String[] s = args.split(" ");
+
+		if (isEvent(s)) {
+			return "event";
+		}
+		
+		if (isDeadline(s)) {
+			return "deadline";
+		}
+		
+		if (isFloat(s)) {
+			return "float";
+		}
+
+		return null;
+	}
+
+	public int wordCount(String s) {
+		if (s == null) {
+			return 0;
+		} else {
+			return s.trim().split("\\s+").length;
+		}
+	}
+	
+	/**
+	 * Checks if arguments contains `to`, at least one LocalDate or LocalTime before `to` and at least one LocalDate or LocalTime after `to`. 
+	 * @param args
+	 * @return true if task is an event
+	 */
+	private boolean isEvent(String[] args) {
+		boolean passCheck = false;
+		// max no. of args for event is 6 (name, sd, st, ed, et, loc)
+		if (args.length <= 6) {
+			passCheck = true;
+		}
+		int toIndex = 0;
+		// check for `to`
+		for (int i = 0; i < args.length; i++) {
+			if (args[i].equals("to")) {
+				passCheck = true;
+				toIndex = i;
+				break;
+			}
+		}
```

Comment:
Seems like `Arrays.toList(args).indexOf("to")` will do the same thing with less code?